File: Leetcode/198. House Robber/solution.go
func rob(nums []int) int {
    a, b, temp := 0, 0, 0
    
    for _, num := range(nums) {
        if a + num > b {
            temp = a + num
        } else {
            temp = b
        }
        a = b
        b = temp
    }
    
    return b
}

File: Leetcode/198. House Robber/solution.rs
impl Solution {
    pub fn rob(nums: Vec<i32>) -> i32 {
        let (mut a, mut b, mut temp) = (0, 0, 0);

        for num in &nums {
            temp = if num + a > b { num + a } else { b };
            a = b;
            b = temp;
        }
        b
    }
}
